### PR TITLE
Add native module loading

### PR DESCRIPTION
**Pull request by Frityet**
Also in this commit, I made it compile for `-Os`, which makes more sense in the context of the web, in order to speed up page load time

### CODE REVIEW

File: fsroot/lua/init.lua [hunk 1]
```diff
@@ -140,8 +140,12 @@ local function fetch_searcher(extension, is_native)
 			local text = fetch_text(test_path)
 			if text then
 				if is_native then
-					error("Cannot load native modules from the web yet")
-					-- return package.loadlib(test_path, "luaopen_"..module:gsub("%.", "_"))
+					local lib, err = package.loadlib(test_path, "luaopen_"..module:gsub("%.so$", ""):gsub("%.", "_"))
+					if lib then
+						return lib
+					else
+						return "Failed to load lib "..test_path..": "..(err or "Unknown error")
```
Comment (Doridian) [inline fsroot/lua/init.lua:147]: This should call `error`, not return a string

**Comment by Doridian** [comment]
> Also in this commit, I made it compile for `-Os`

I mean, on initial page load, sure. But it should get cached pretty quickly and with HTTP compression (gzip) it is still way smaller than most modern web apps.

In either case, since the real slowdown comes from the glue layer between JS and Lua, I doubt `-O3` would even gain us much (but I also doubt `-Os` does much?).

Just testing it out `-Os` reduces our size by a total of like 150kB, so I don't think it is worth it honestly, especially with a well configured cache.

See sizes below:

| `-O` | `gzip` | `luajs.js` size | `luajs.wasm` size |
| --- | --- | --- | --- |
| `-O3` | ❌  | 104 kB | 590 kB |
| `-Os` |  ❌ | 104 kB | 432 kB |
| `-O3` | ✅ | 28 kB | 195 kB |
| `-Os` |  ✅ | 28 kB | 150 kB |

Also I just fixed the `sed` scripts being entirely broken, sorry about the merge conflicts.

**Comment by Frityet** [comment]
>I mean, on initial page load, sure. But it should get cached pretty quickly and with HTTP compression (gzip) it is still way smaller than most modern web apps.
In either case, since the real slowdown comes from the glue layer between JS and Lua, I doubt -O3 would even gain us much (but I also doubt -Os does much?).
Just testing it out -Os reduces our size by a total of like 150kB, so I don't think it is worth it honestly, especially with a well configured cache.

Makes sense, even then, maybe consider changing it to `-O2`? `-O2` can often generate faster code, with less size, whilst also being less harsh if there is UB (https://mattermodeling.stackexchange.com/questions/9470/why-does-vasp-use-o2-optimization-by-default-rather-than-o3)

File: fsroot/lua/init.lua (hunk 1 above)
Comment (Frityet) [reply fsroot/lua/init.lua:147]: I intentionally return a string just in case a user-defined loader that comes after can properly load it

**Comment by Doridian** [comment]
@Frityet Feel free to fix your merge conflicts (I assume they should be fairly minimal if you remove your `-O#` flag changes.

I have also switched to `-O2` because after doing some reading, I agree `-O3` might be a bit much.

**Comment by Frityet** [comment]
Will do! give me a few days

**Pull request closed without merging**